File: sdk/java-sdk/src/test/java/com/evernym/verity/sdk/TestHelpers.java
package com.evernym.verity.sdk;

import com.evernym.verity.sdk.exceptions.WalletException;
import com.evernym.verity.sdk.utils.Context;
import com.evernym.verity.sdk.utils.ContextBuilder;
import com.evernym.verity.sdk.utils.TestWallet;
import com.evernym.verity.sdk.wallet.WalletConfig;
import org.hyperledger.indy.sdk.wallet.Wallet;
import org.json.JSONObject;

import java.util.UUID;

import static com.evernym.verity.sdk.utils.Util.unpackMessage;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class TestHelpers {
    protected TestHelpers() {
    }

    public static Context getContext() throws WalletException {
        return getContext(null);
    }

    public static Context getContext(String seed) throws WalletException {

        String walletName = UUID.randomUUID().toString();
        String walletKey = UUID.randomUUID().toString();
        String endpointUrl = "http://localhost:3000";
        String verityUrl = "http://localhost:3000";

        TestWallet testWallet = new TestWallet(walletName, walletKey, seed);
        return ContextBuilder
                .blank()
                .walletConfig(testWallet)
                .verityUrl(verityUrl)
                .verityPublicDID(testWallet.getVerityPublicVerkey())
                .verityPublicVerKey(testWallet.getVerityPublicVerkey())
                .domainDID(testWallet.getVerityPairwiseDID())
                .verityAgentVerKey(testWallet.getVerityPairwiseVerkey())
                .sdkVerKeyId(testWallet.getSdkPairwiseDID())
                .sdkVerKey(testWallet.getSdkPairwiseVerkey())
                .endpointUrl(endpointUrl)
                .build();
    }

    public static void cleanup(Context context) throws Exception {
        if(context != null) {
            if(! context.walletIsClosed()) {
                context.closeWallet();
            }
            WalletConfig config = context.walletConfig();
            Wallet.deleteWallet(config.config(), config.credential()).get();
        }
    }

    public static void assertEqualsJSONObject(JSONObject expected, JSONObject actual) {
        assertEquals(expected.toMap().size(), actual.toMap().size());
        expected.toMap().forEach((s, o) ->
                assertEquals(actual.get(s), o)
        );
    }


    public static JSONObject unpackForwardMessage(Context context, byte[] message) throws WalletException {
        JSONObject unpackedOnceMessage = unpackMessage(context, message);
        byte[] unpackedOnceMessageMessage = unpackedOnceMessage.getJSONObject("@msg").toString().getBytes();
        return unpackMessage(context, unpackedOnceMessageMessage);
    }
}

File: sdk/dotnet-sdk/src/VeritySDK.Test/TestHelpers.cs
using Com.Evernym.Vdrtools.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Json;
using System.Text;
using VeritySDK.Utils;
using VeritySDK.Wallets;

namespace VeritySDK.Test
{
    public class TestHelpers
    {
        protected TestHelpers()
        {
        }

        public static Context getContext()
        {
            return getContext(null);
        }

        public static Context getContext(string seed)
        {
            var env = new Env();

            TestWallet testWallet = new TestWallet(env.walletName, env.walletKey, seed);
            return ContextBuilder
                    .blank()
                    .walletConfig(testWallet)
                    .verityUrl(env.verityUrl)
                    .verityPublicDID(testWallet.getVerityPublicVerkey())
                    .verityPublicVerKey(testWallet.getVerityPublicVerkey())
                    .domainDID(testWallet.getVerityPairwiseDID())
                    .verityAgentVerKey(testWallet.getVerityPairwiseVerkey())
                    .sdkVerKeyId(testWallet.getSdkPairwiseDID())
                    .sdkVerKey(testWallet.getSdkPairwiseVerkey())
                    .endpointUrl(env.endpointUrl)
                    .build();
        }


        public static void assertEqualsJSONObject(JsonObject expected, JsonObject actual)
        {
            Assert.AreEqual(expected.Count, actual.Count);

            foreach (var exp in expected)
            {
                var act_v = actual[exp.Key].ToString();
                var exp_v = exp.Value.ToString();

                Assert.AreEqual(act_v, exp_v);
            }
        }

        public static void cleanup(Context context)
        {
            if (context != null)
            {
                if (!context.WalletIsClosed())
                {
                    context.CloseWallet();
                }
                WalletConfig config = context.WalletConfig();
                Wallet.DeleteWalletAsync(config.config(), config.credential()).GetAwaiter().GetResult();
            }
        }

        public static JsonObject unpackForwardMessage(Context context, byte[] message)
        {
            JsonObject unpackedOnceMessage = Util.unpackMessage(context, message);
            byte[] unpackedOnceMessageMessage = Encoding.UTF8.GetBytes(unpackedOnceMessage["@msg"].ToString());
            return Util.unpackMessage(context, unpackedOnceMessageMessage);
        }

    }
}
